arduino-ide-extension/src/browser/contributions/check-for-ide-updates.ts

```diff
@@ -66,4 +85,26 @@ export class CheckForIDEUpdates extends Contribution {
         );
       });
   }
+
+  private async setLastUsedIDEVersion(force = false): Promise<void> {
+    const { appVersion } = await this.appService.info();
+    const lastUsedIDEVersion = await this.localStorage.getData<string>(
+      LAST_USED_IDE_VERSION
+    );
+    if (validSemVer(appVersion) && (!lastUsedIDEVersion || force)) {
+      this.localStorage.setData(LAST_USED_IDE_VERSION, appVersion);
+    }
+  }
+
+  private async isRunningNewIDEVersion(): Promise<boolean> {
+    const { appVersion } = await this.appService.info();
+    const prevVersion = await this.localStorage.getData<string>(
+      LAST_USED_IDE_VERSION
+    );
+    try {
+      return !!prevVersion && new SemVer(appVersion).compare(prevVersion) === 1;
```

Comment:
```suggestion
      const appVersionObject = new SemVer(appVersion);
      return (
        !!prevVersion &&
        appVersionObject.prerelease.length === 0 &&
        appVersionObject.compare(prevVersion) === 1
      );
```

I think it is better to suppress the dialog when the user is using a pre-release version. It is sufficient to show it only after updates to production releases.

The previous code will cause the appearance of the dialog to occur intermittently for users of the [tester builds](https://github.com/arduino/arduino-ide/blob/main/docs/contributor-guide/beta-testing.md#testing-pull-requests), since these builds use the format `X.Y.Z-snapshot-<commit hash>`, and a tester build version may be considered higher or lower than another version at random purely based on the [lexicographical relationship](https://semver.org/#spec-item-11) between the two commit hashes. This apparent inconsistency might cause beta testers to think there is a bug in the system.

The previous code will cause the appearance of the dialog every time the user updates to a new [nightly build](https://github.com/arduino/arduino-ide/blob/main/docs/contributor-guide/beta-testing.md#testing-nightly-build) (which a dedicated beta tester will do frequently).

The prerelease builds are primarily used by beta testers. [Beta testing is an valuable contribution](https://github.com/arduino/arduino-ide/blob/main/docs/contributor-guide/beta-testing.md#beta-testing-guide) already so I don't think it is fair to subject these users to more excessive donation requests than to those who only use release builds.

I'm not skilled in writing TypeScript so the code I suggest here might be suboptimal. I provide it more to clearly define the nature of my proposal. Feel free to implement the code anyway you like if my proposal is accepted.